R/jk2.R

```diff
@@ -329,7 +329,28 @@ eatRep <- function (datL, ID, wgt = NULL, L1wgt=NULL, L2wgt=NULL, type = c("none
           separate.missing.indicator = FALSE, expected.values = NULL, probs = NULL, nBoot = NULL, bootMethod = NULL, formula=NULL, family=NULL, formula.fixed=NULL, formula.random=NULL,
           forceSingularityTreatment = FALSE, glmTransformation = c("none", "sdY"), correct, onlyCheck = FALSE, modus, poolMethod = "mice", useWec = FALSE, engine,
           scale, rscales, mse, rho, reihenfolge = NULL, hetero, se_type, crossDiffSE.engine, stochasticGroupSizes, verbose, progress, clusters=NULL, fc = NULL, isRecursive = FALSE, depOri = NULL) {
-          datL  <- eatTools::makeDataFrame(datL, name = "datL", minRow = 2, onlyWarn=FALSE)
+  
+  # checks/assertions
+  datL  <- eatTools::makeDataFrame(datL, name = "datL", minRow = 2, onlyWarn=FALSE)
+  if (checkmate::test_numeric(ID)){
+    checkmate::assert_numeric(ID, len = 1, any.missing = FALSE)
+  } else {checkmate::assert_character(ID, len = 1, any.missing = FALSE)}
+  lapply(c(cross.differences, useEffectLiteR, na.rm, forcePooling, doCheck,
+           separate.missing.indicator, forceSingularityTreatment, onlyCheck, useWec, isRecursive), 
+         assert_logical, len = 1)
+  
+  if(checkmate::test_character(wgt)){
+    checkmate::assert_character(wgt, len = 1, null.ok = TRUE)
+  } else{checkmate::assert_numeric(wgt, len = 1, null.ok = TRUE)}
+  lapply(c(L1wgt, L2wgt), checkmate::assert_character, len = 1, null.ok = TRUE)
+  if(checkmate::test_character(PSU)){
+    checkmate::assert_character(PSU, len = 1, null.ok = TRUE)
+  } else{checkmate::assert_numeric(PSU, len = 1, null.ok = TRUE)}
+  if(checkmate::test_character(repInd)){
+    checkmate::assert_character(repInd, len = 1, null.ok = TRUE)
+  } else{checkmate::assert_numeric(repInd, len = 1, null.ok = TRUE)}

```

Comment:
I haven't tried it out, but can you also do that with lappy or a similar function? because, unfortunately you can't check for 2 types at the same time. I could write assertions just checking for scalar input:
> lapply(c(wgt, PSU, repInd, ...), checkmate::assert_scalar, null.ok = TRUE)
this would allow logical input as well, but i would make the code simpler/easier to read.

Reply:
All arguments that refer to columns or column names in the data set (`ID`, `PSU`, `repInd`, `wgt`, ...) are checked using the check function `eatTools::existsBackgroundVariables`. In my opinion, no additional assertions are necessary for these.